src/cmap/handshake/client_metadata.ts

```diff
@@ -0,0 +1,94 @@
+import * as os from 'os';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const NODE_DRIVER_VERSION = require('../../../package.json').version;
+
+/**
+ * FaaS environment metadata.
+ * @public
+ */
+export interface FaasMetadata {
+  /** All metadata has a name */
+  name: string;
+  /** Lambda/GCP/Vercel */
+  region?: string;
+  /** Lambda/GCP */
+  memoryMb?: string;
+  /** GCP */
+  timeoutSec?: string;
+  /** Vercel */
+  url?: string;
+}
+
+/** @public */
+export interface ClientMetadata {
+  driver: {
+    name: string;
+    version: string;
+  };
+  os: {
+    type: string;
+    name: NodeJS.Platform;
+    architecture: string;
+    version: string;
+  };
+  platform: string;
+  version?: string;
+  application?: {
+    name: string;
+  };
+  env?: FaasMetadata;
+}
+
+/** @public */
+export interface ClientMetadataOptions {
+  driverInfo?: {
+    name?: string;
+    version?: string;
+    platform?: string;
+  };
+  appName?: string;
+}
+
+export function makeClientMetadata(options?: ClientMetadataOptions): ClientMetadata {
+  options = options ?? {};
```

Comment:
```suggestion
export function makeClientMetadata(options: ClientMetadataOptions = {}): ClientMetadata {
```

might as well use default arg here.